### ISSUE TITLE
Identify cases when browser fails to establish a connection

### ISSUE THREAD

**Issue by wentwrong** [DevExpress/testcafe#5683]
If launched browser didn't visit a connection URL during the timeout period, the following error message is shown:
> ERROR Unable to establish one or more of the specified browser connections. This can be caused by network issues or remote device failure.

We should show a more specific error message in some of the cases when this error occurs. Listed below are a few of the cases in which the error can be thrown:

- [x] When browser takes too long to initialize (See: [Improve the error message when browser takes a too long time to initialize](https://github.com/DevExpress/testcafe/issues/5307)). This case is very common when concurrency mode is used, especially in CPU-restricted environments.

- [ ] When browser failed to start. Our current approach to browser launching has some drawbacks - for one, we're ignoring stderr. For example, when using "ubuntu" docker image, chromium browser writes the following to the standard error stream:
```
Command '/usr/bin/chromium-browser' requires the chromium snap to be installed.        
Please install it with:

snap install chromium
```
This information can be useful to the user, but we never read it. The stdin redirection also [introduces](https://github.com/DevExpress/testcafe/issues/1640#issuecomment-719399471) a problem - we can't start mounted Windows browser executables in WSL.

- [ ] When there's a network problem and browser was unable to open the connection URL

**Comment by Michal04** [comment]
I had this issue recently on Jenkins:

""ERROR Unable to establish one or more of the specified browser connections. This can be caused by network issues or remote device failure.""

Testcafe does not show why and I spent a day trying to find out. I gave up and installed Cypress. And to my surprise, they show the error:

"/var/lib/jenkins/.cache/Cypress/3.8.3/Cypress/Cypress: error while loading shared libraries: libgtk-3.so.0: cannot open shared object file: No such file or directory"

so when i installed "apt-get install libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb", it solved Testcafe issue too.
